calculator.js

```diff
@@ -1,73 +1,146 @@
-      final = calculation();
-      count.display.innerHTML = `${final}`;
-  } else if (character == DELETE) {
-      count.firstTerm = EMPTYsTRING;
-      count.secondTerm = EMPTYsTRING;
-      count.term = EMPTYsTRING;
-      count.display.innerHTML = DISPLAYzERO;
-  }
+const saveNumber = function(event) {  
+  const character = event.target.value;
+  calculator.term += character;
+  calculator.current += character;
+  livesResult.innerHTML = calculator.current;
 }
 /*
- *PRE: It took a bunch of elements from the DOM in this case inputs, to add a event listener to them. Because it is a calculator. 
- *POST: If de event is true, It executes process to get a solution.   
+ *PRE-CONDITIONS: This function receive two variables that are going to be used to change two global variables.
+ *POST-CONDITIONS: Change the state of the display.
  */
-OUTCOME.forEach(function(input){
-  input.addEventListener("click", SAVEcHARACTER)
+function showDisplay(stringOperations, total) {
+  display.innerHTML = stringOperations;
+  livesResult.innerHTML = total;
+}
+/*
+ *PRE-CONDITIONS: Receive one variable.
+ *POST-CONDITIONS: Change the state of two variables.
+ */
+function reassignedVariables(operator) {
+  calculator.current = "";
+  calculator.operator = operator;
+}
+/*
+ *PRE-CONDITIONS: This function gets information from global variables filled previously.
+ *POST-CONDITIONS: It Calculates a number.
+ */
+const operate = function (event) {
+  if (!calculator.operator) {
+    calculator.operator = event.target.value;
+    calculator.nextOperator = calculator.operator;
+    calculator.term += calculator.operator;
+    calculator.total = calculator.current;
+    reassignedVariables(calculator.nextOperator);
+    showDisplay(calculator.term, calculator.total);
+  } else {
+    calculator.nextOperator = event.target.value;
+    calculator.term += calculator.nextOperator;
+    calculator.total = calculation(calculator.total, calculator.current);
+    reassignedVariables(calculator.nextOperator);
+    showDisplay(calculator.term, calculator.total);
+  }
+};
+/*
+ *PRE-CONDITIONS: This function gets information from global variables filled previously.
+ *POST-CONDITIONS: It does three actions when it activates by event listener. 
+ */
+const doActions = function (event) {
+  const character = event.target.value; 
+  if(character === EQUAL) {
+    calculator.total = calculation(calculator.total, calculator.current);
+    livesResult.innerHTML = calculator.total;
+    display.innerHTML = "0";   
+    calculator.current = 0;
+  }
+  if (character === CLEAR) {
+    calculator.current = "";
+    calculator.operator = "";
+    calculator.nextOperator = "";
+    calculator.term = "";
+    calculator.total = 0;
+    display.innerHTML = 0;
+    livesResult.innerHTML = 0;
+  }
+  if (character === DELETE) {
+    const firstTerm = calculator.term; 
+    const number = calculator.current;
+    const box = number.slice(0,-1); 
+    const boxOne = firstTerm.slice(0,-1);
```

Comment:
`box` and `boxOne` are not reused. Assign them directly to `term` and `current`.